pkg/accountmanager/accountmanageriface/servicer.go

```diff
@@ -5,14 +5,26 @@ package accountmanageriface
 import (
 	"github.com/Optum/dce/pkg/account"
 	"github.com/Optum/dce/pkg/arn"
+	"github.com/aws/aws-sdk-go/aws/credentials"
 )
 
 // Servicer makes working with the Account Manager easier
+//go:generate mockery -name Servicer
 type Servicer interface {
 	// ValidateAccess creates a new Account instance
 	ValidateAccess(role *arn.ARN) error
+	// Retrieve credentials for the provided IAM Role ARN
+	Credentials(role *arn.ARN, roleSessionName string) Credentialer
+	// ConsoleURL generates a URL that may be used
+	// to login to the AWS web console for an account
+	ConsoleURL(creds Credentialer) (string, error)
```

Comment:
Figured the AccountManager service was a good place for this abstraction, as it already has logic for managing assume-role creds